### PR TITLE
[Hotfix][Connector-V2][AssertSink]Fix the problem that the data type

### LINKED ISSUE #4635: [Bug] [Assert Sink]Fix the problem that the data type
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

I defined an Assertsink, using the field_rules method to validate data. I found that bigint, bytes, etc. are not supported. I checked the documentation, and the types do not correspond with some parts of the ST Row type. Please translate this into English.

I have defined an Assertsink and used the field_rules method for data validation. I discovered that bigint, bytes, and so on are not supported. I've reviewed the documentation, and the types don't fully match up with some parts of the ST Row type.

图片

### SeaTunnel Version

v2.3.1

### SeaTunnel Config

```conf
sink {
  Assert {
    source_table_name = "fake1"
    rules =
      {
        row_rules = [
          {
            rule_type = MAX_ROW
            rule_value = 100
          },
          {
            rule_type = MIN_ROW
            rule_value = 5
          }
        ],
        field_rules = [{
          field_name = name
          field_type = string
          field_value = [
            {
              rule_type = NOT_NULL
            },
            {
              rule_type = MIN_LENGTH
              rule_value = 5
            },
            {
              rule_type = MAX_LENGTH
              rule_value = 65535
            }
          ]
        }, {
          field_name = age
          field_type = bigint
          field_value = [
            {
              rule_type = NOT_NULL
            },
            {
              rule_type = MIN
              rule_value = 32767
            },
            {
              rule_type = MAX
              rule_value = 2147483647
            }
          ]
        }
        ]
      }

  }
```

### Running Command

```shell
null
```

### Error Exception

```log
<img width="1188" alt="图片" src="https://user-images.githubusercontent.com/60029759/233310035-7a760d1a-fd79-49e9-9a1e-73e09c964e6c.png">
```

### Flink or Spark Version

_No response_

### Java or Scala Version

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by MonsterChenzhuo**
## Purpose of this pull request
close #4635

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by MonsterChenzhuo** [comment]
图片
图片
I don't know what caused the failure.

**Comment by TyrantLucifer** [comment]
Because assert sink is a basic component for connectors, there are many test cases involved. If you want to update parameters, please update all config files too. This is not a prod connector so there is little risk of compatibility, and I see anyone submit another issue #4566, IMO we should refactor this connector and make this connector more powerful. Hey guys, what do you think about it? @EricJoy2048 @hailin0 @Hisoka-X

**Comment by hailin0** [comment]
> If you want to update parameters, please update all config files too.

+1

File: seatunnel-connectors-v2/connector-assert/src/main/java/org/apache/seatunnel/connectors/seatunnel/assertion/rule/AssertRuleParser.java [hunk 1]
```diff
@@ -96,11 +96,15 @@ private SeaTunnelDataType<?> getFieldType(String fieldTypeStr) {
         TYPES.put("boolean", BasicType.BOOLEAN_TYPE);
         TYPES.put("byte", BasicType.BYTE_TYPE);
         TYPES.put("short", BasicType.SHORT_TYPE);
+        TYPES.put("tinyint", BasicType.BYTE_TYPE);
+        TYPES.put("smallint", BasicType.SHORT_TYPE);
         TYPES.put("int", BasicType.INT_TYPE);
+        TYPES.put("bigint", BasicType.LONG_TYPE);
         TYPES.put("long", BasicType.LONG_TYPE);
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-assert/src/main/java/org/apache/seatunnel/connectors/seatunnel/assertion/rule/AssertRuleParser.java:103]: update all config files and remove this option

**Pull request closed without merging**